contribute/architecture/state.md

```diff
@@ -0,0 +1,95 @@
+# State
+
+Grafana persists state in a SQLite, MySQL, or PostgreSQL database. 
+
+Grafana uses the [XORM](https://xorm.io) framework for persisting objects to the database. For more information, refer to the [XORM documentation](http://gobook.io/read/github.com/go-xorm/manual-en-US/).
+
+There are two ways of persisting state in Grafana: using `sqlstore` handlers, and using the `SqlStore` instance.
+
+## `sqlstore` handlers
+
+> **Deprecated:** We are deprecating `sqlstore` handlers in favor of using the `SqlStore`  object directly in each service. Since most services still use the `sqlstore` handlers, we still want to explain how they work.
+
+In short, `sqlstore` handlers are [command handlers](services.md#handle-commands) that persist state to a database. 
```

Comment:
There's query handlers as well.

Reply:
A query is really a command where you update one or more field on the command being handled. Do we need to treat them as two distinct things here?